lib/libvarnish/binary_heap.c

```diff
@@ -244,6 +244,21 @@ binheap_new(void *priv, binheap_cmp_t *cmp_f, binheap_update_t *update_f)
 	return (bh);
 }
 
+void
+binheap_destroy(struct binheap **bhp)
+{
+	struct binheap *bh;
+	unsigned u;
+
+	TAKE_OBJ_NOTNULL(bh, bhp, BINHEAP_MAGIC);
+	AZ(binheap_root(bh));
```

Comment:
Why should it be forbidden to release a non-empty binheap?

Reply:
my thinking was that the binheap might be the only reference and a delete is likely to require more than just the binheap delete

Reply:
Good point!